cmd/wstail/main.go

```diff
@@ -1,34 +1,60 @@
 package main
 
 import (
+	"context"
 	"flag"
 	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
 
 	log "github.com/Sirupsen/logrus"
 	"github.com/gravitational/trace"
 )
 
+const filePathContextKey = "filepath"
+
 func main() {
 	log.SetLevel(log.InfoLevel)
+
+	var (
+		filePath string
+		httpAddr = flag.String("addr", ":8083", "HTTP service address")
+	)
 	flag.Parse()
 	if flag.NArg() < 1 {
 		filePath = defaultTailSource
 	} else {
 		filePath = flag.Args()[0]
 	}
 
-	http.Handle("/v1/ws", makeHandler(serveWs))
-	http.Handle("/v1/download", makeHandler(downloadLogs))
+	log.Infof("HTTP service listening on %s", *httpAddr)
+
+	http.Handle("/v1/log", withContext(filePathContextKey, filePath, makeHandler(getLogs)))
```

Comment:
Sure, it's possible, but less flexible. I used context as general-purpose container with versatile middleware, you can set anything later without rewriting function signatures